cmake/modules/FindSUNDIALS.cmake

```diff
@@ -0,0 +1,58 @@
+# =============================================================================
+# Sundials
+# =============================================================================
+
+include(ExternalProject)
+set(SUNDIALS_PREFIX "${THIRD_PARTY_DIRECTORY}/sundials")
+set(SUNDIALS_INCLUDE_DIR "${SUNDIALS_PREFIX}/include")
+set(SUNDIALS_LIB_DIR "${SUNDIALS_PREFIX}/lib")
+make_directory(${SUNDIALS_INCLUDE_DIR})
+
+set(SUNDIALS_C_FLAGS ${CMAKE_C_FLAGS})
+list(REMOVE_ITEM SUNDIALS_C_FLAGS "-tp=haswell")
+set(SUNDIALS_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+list(REMOVE_ITEM SUNDIALS_CXX_FLAGS "-tp=haswell")
+include(ExternalProject)
+ExternalProject_Add(
+  sundials-external
+  PREFIX "${SUNDIALS_PREFIX}"
+  # cmake-format: off
+  # GIT_REPOSITORY https://github.com/LLNL/sundials.git
+  # Forked to following, to allow RPowerR to use hoc_pow
+  GIT_REPOSITORY https://github.com/neuronsimulator/sundials.git
+  # GIT_TAG e2f29c34f324829302037a1492db480be8828084   6.2.0 -> CVodeMem no longer "visible" GIT_TAG
+  # c09e732080a214694b209032ec627c93fed45340  4
+  #GIT_TAG 811234254d37652954daff0ccdb7af9813736846 # 3.2.1
+  GIT_TAG 71b3daec18ce66a7c011be9501a31567aa5c09b6 # 3.2.1 hines/hoc_pow
+  # GIT_TAG 7ed895fd102226dbc52225e6b2c6e26ef1cafa0e  #2.7.0
+  # cmake-format: on
```

Comment:
@nrnhines : is this https://github.com/LLNL/sundials/compare/main...neuronsimulator:sundials:hines/hoc_pow only change that we need to have? Indeed it's look very trivial.

Reply:
That is correct for Sundials version 3.  We'll eventually see if changes need to be more substantial for later versions. There is some question about where to put the 10 nrn/src/nrniv/nvector_* files. They distinguish between (serial, thread, mpi) and (double, longdouble) and I don't think the changes are relevant to sundials as they have details that depend on how NEURON uses threads and mpi and NRN_ENABLE_MPI_DYNAMIC. At the moment, when
NRN_ENABLE_MPI_DYNAMIC=ON, then ```nm libnrniv.so | grep MPI``` shows nothing that points into libmpi. Instead, many things point into mpi specific interface libraries such as libnrnmpi_mpich.so and libnrnmpi_ompi.so